algorithms/src/snark/varuna/tests.rs

```diff
@@ -524,3 +524,293 @@ mod varuna_hiding {
+        let instance_file = fs::read_to_string(instance_path).expect("Could not read the file");
+        let mut instance = Vec::new();
+        for line in instance_file.lines() {
+            instance.push(line.to_string())
+        }
+
+        // Initialize challenges from file.
+        let challenges_path = "src/snark/varuna/resources/challenges.input";
+        let challenges_file = fs::read_to_string(challenges_path).expect("Could not read the file");
+        let mut challenges = Vec::new();
+        for line in challenges_file.lines() {
+            challenges.push(line)
+        }
+        let (alpha, _eta_a, eta_b, eta_c, beta, delta_a, delta_b, delta_c, _gamma) = (
+            Fr::from_str(challenges[0]).unwrap(),
+            Fr::from_str(challenges[1]).unwrap(),
+            Fr::from_str(challenges[2]).unwrap(),
+            Fr::from_str(challenges[3]).unwrap(),
+            Fr::from_str(challenges[4]).unwrap(),
+            vec![Fr::from_str(challenges[5]).unwrap()],
+            vec![Fr::from_str(challenges[6]).unwrap()],
+            vec![Fr::from_str(challenges[7]).unwrap()],
+            Fr::from_str(challenges[8]).unwrap(),
+        );
+        let circuit_combiner = Fr::one();
+        let instance_combiners = vec![Fr::one()];
+
+        let mut random_state = vec![0u64; 100];
+        let add_f_to_state = |s: &mut Vec<u64>, f: Fp256<FrParameters>| {
+            // Fp384 field elements sample 6 u64 values in total.
+            for i in 0..f.0.0.len() {
+                s.push(f.0.0[f.0.0.len() - 1 - i]);
+            }
+        };
+        for witness in instance[15].split(',') {
+            if witness.trim() == "" {
+                continue;
+            }
+            add_f_to_state(&mut random_state, Fr::from_str(witness.trim()).unwrap());
+        }
+
+        let rng = &mut snarkvm_utilities::rand::TestMockRng::fixed(random_state);
+
+        let max_degree = AHPForR1CS::<Fr, MM>::max_degree(100, 25, 300).unwrap();
+        let universal_srs = VarunaSonicInst::universal_setup(max_degree).unwrap();
+        let mul_depth = 3;
+        let num_constraints = 7;
+        let num_variables = 7;
+
+        // TODO: pass the right randomness in
+        let (circ, _) = TestCircuit::gen_rand(mul_depth, num_constraints, num_variables, rng);
+        let (index_pk, _index_vk) = VarunaSonicInst::circuit_setup(&universal_srs, &circ).unwrap();
+        let mut keys_to_constraints = BTreeMap::new();
+        keys_to_constraints.insert(index_pk.circuit.deref(), std::slice::from_ref(&circ));
+
+        let prover_state = AHPForR1CS::<_, MM>::init_prover(&keys_to_constraints, rng).unwrap();
+        let mut prover_state = AHPForR1CS::<_, MM>::prover_first_round(prover_state, rng).unwrap();
+        let first_round_oracles = Arc::new(prover_state.first_round_oracles.as_ref().unwrap());
+
+        // Get private witness polynomial coefficients.
+        // TODO: Support batched circuits
```

Comment:
Remove these and the same TODOs below, added here https://github.com/orgs/AleoHQ/projects/13?pane=issue&itemId=40044078